ui/src/app/views/workspace-edit.tsx

```diff
@@ -505,6 +586,11 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
       return fp.includes(populationEnum, this.state.workspace.researchPurpose.populationDetails);
     }
 
+    toggleDiseaseHover(diseaseIndex) {
+      this.state.diseaseHover[diseaseIndex] = !this.state.diseaseHover[diseaseIndex];
```

Comment:
In general, we avoid directly setting state using `=`, and instead use the `setState` method, because it does the re-rendering under the hood. And both times you use this method, it is used within a setState call. I think you could even do something pretty like:
`this.setState({diseaseHover: fp.setWith(Array, '['+diseaseIndex+']', !this.state.diseaseHover[diseaseIndex], this.state.diseaseHover)` if you wanted to one-line it. (Not sure this is as readable? Also, not sure on syntax. But generally avoid direct `=` setting on state.)

Reply:
It seems that `fp.setWith(Array, '['+diseaseIndex+']', !this.state.diseaseHover[diseaseIndex], this.state.diseaseHover)` and `toggleDiseaseHover` are functionally equivalent, except that with `fp.setWith` the array indexing code is interpreted rather than compiled. If there's no simpler way to express assignment then I don't care to change this. Re-rendering will happen either way since `toggleDiseaseHover()` is evaluated within the call to `setState`.

Reply:
I would rather use the setState in here rather than using a `state = {new state}` then. Because nothing with this function says that it needs to be called within a setState, and I'd rather have this in code. It seems fine not to use the setWith (I think its less readable, and this is fine), but definitively doing a setState here ensures it can't be misused.